tests/units/cli/test_commands.py

```diff
@@ -662,3 +678,202 @@ def test_run_command(self):
+            {'config': 'valid', 'properties': 'invalid', 'state': 'valid'},
+            {'config': 'valid', 'properties': 'valid', 'state': 'invalid'},
+        )
+        for json_files_situation in test_cases:
+            self._assert_retry_validation_of_json_file(json_files_situation)
+
+    def test_tap_config_jason_valid_if_state_file_does_not_exist(self):
+        """Test it is valid if state file does not exists at all"""
+        with TemporaryDirectory() as temp_dir:
+            valid_json_file = f'{temp_dir}/valid_file.json'
+            not_exists_state_file = f'{temp_dir}/not_exists.json'
+            with open(valid_json_file, 'w', encoding='utf-8') as valid_file:
+                json.dump({'foo': 'bar'}, valid_file)
+
+            self._assert_tap_config(config=valid_json_file, properties=valid_json_file, state=not_exists_state_file)
+
+    # pylint: disable=invalid-name
+    def test_tap_config_json_valid_if_state_file_is_empty_and_raise_exception_on_invalid_content(self):
+        """Test it is valid if state file is empty and raise exception if invalid content"""
+        invalid_file_contents = [' ', 'foo', '{"foo": 1']
+
+        with TemporaryDirectory() as temp_dir:
+            invalid_json_file = f'{temp_dir}/invalid_file.json'
+            valid_json_file = f'{temp_dir}/valid_file.json'
+            empty_file = f'{temp_dir}/empty_stat.json'
+
+            with open(valid_json_file, 'w', encoding='utf-8') as valid_file:
+                json.dump({'foo': 'bar'}, valid_file)
+
+            # 1. asserting if raises exception on invalid contents
+            for invalid_content in invalid_file_contents:
+                with open(invalid_json_file, 'w', encoding='utf-8') as invalid_file:
+                    invalid_file.write(invalid_content)
+
+                with pytest.raises(commands.RunCommandException) as command_exception:
+                    self._assert_tap_config(
+                        config=valid_json_file,
+                        properties=valid_json_file,
+                        state=invalid_json_file,
+                    )
+                assert str(command_exception.value) == f'Invalid json file for state: {invalid_json_file}'
+
+            # 2. asserting if it is valid with empty file
+            with open(empty_file, 'w', encoding='utf-8') as state_file:
+                state_file.write('')
+            self._assert_tap_config(
+                config=valid_json_file,
+                properties=valid_json_file,
+                state=empty_file
+            )
+
+    def test_tap_config_valid_if_json_property_is_none(self):
+        """Test TapConfig is valid if a json property is None"""
+        with TemporaryDirectory() as temp_dir:
+            valid_json_file = f'{temp_dir}/valid_file.json'
+            with open(valid_json_file, 'w', encoding='utf-8') as valid_file:
+                json.dump({'foo': 'bar'}, valid_file)
+
+            test_cases = (
```

Comment:
please break this into separate test cases(methods)

Reply:
- [ ] Make separate test cases